apis/secretsmanager/v1alpha1/common.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+)
+
+// GenerateSecretTags generates a tag array with type that secretsmanager client expects.
+func GenerateSecretTags(tags []Tag) []secretsmanager.Tag {
+	res := make([]secretsmanager.Tag, len(tags))
+	for i, t := range tags {
+		res[i] = secretsmanager.Tag{Key: aws.String(t.Key), Value: aws.String(t.Value)}
+	}
+	return res
+}
+
+// TagsToMap converts []secretsmanager.Tag to map
+func TagsToMap(tags []Tag) map[string]string {
+	result := make(map[string]string)
+	for i := range tags {
+		result[tags[i].Key] = tags[i].Value
+	}
+	return result
+}
+
+// SecretsmanagerTagsToMap converts []secretsmanager.Tag to map
+func SecretsmanagerTagsToMap(secretsmanagerTags []secretsmanager.Tag) map[string]string {
+	result := make(map[string]string)
+	for i := range secretsmanagerTags {
+		result[aws.StringValue(secretsmanagerTags[i].Key)] = aws.StringValue(secretsmanagerTags[i].Value)
+	}
+	return result
+}
+
+// DiffTags returns tags that should be added or removed.
+func DiffTags(spec []Tag, current []secretsmanager.Tag) (addTags []secretsmanager.Tag, remove []string) {
+	local := TagsToMap(spec)
+	remote := SecretsmanagerTagsToMap(current)
+	add := make(map[string]string, len(local))
+	remove = []string{}
+	for k, v := range local {
+		add[k] = v
+	}
+	for k, v := range remote {
+		switch val, ok := local[k]; {
+		case ok && val != v:
+			remove = append(remove, k)
+		case !ok:
+			remove = append(remove, k)
+			delete(add, k)
+		default:
+			delete(add, k)
+		}
+	}
+	addTags = []secretsmanager.Tag{}
+	for key, value := range add {
+		value := value
+		key := key
+		addTags = append(addTags, secretsmanager.Tag{Key: &key, Value: &value})
```

Comment:
```suggestion
		addTags = append(addTags, secretsmanager.Tag{Key: awsclients.String(key), Value: awsclients.String(value)})
```

You can to use this helper function here.